protocols/kad/src/behaviour.rs

```diff
@@ -769,20 +892,24 @@ where
                                     NetworkBehaviourAction::GenerateEvent(event));
                             },
                             kbucket::InsertResult::Full => {
-                                debug!("Bucket full. Peer not added to routing table: {}", peer)
+                                debug!("Bucket full. Peer not added to routing table: {}", peer);
+                                let address = addresses.first().clone();
+                                self.queued_events.push_back(NetworkBehaviourAction::GenerateEvent(
+                                    KademliaEvent::RoutablePeer { peer, address }
+                                ));
                             },
                             kbucket::InsertResult::Pending { disconnected } => {
                                 debug_assert!(!self.connected_peers.contains(disconnected.preimage()));
+                                let address = addresses.first().clone();
+                                self.queued_events.push_back(NetworkBehaviourAction::GenerateEvent(
+                                    KademliaEvent::PendingRoutablePeer { peer, address }
```

Comment:
Doc says
>  A connection to a peer has been established

I'm not sure if that's true at this point. `DialPeer` is being scheduled, it may fail. Also, `DialPeer` may result in dialing not `addresses.first()`, but some other address.

Just wanted to state that this is a bit misleading :) Maybe reflect that in the doc comment on `PendingRoutablePeer`?

Reply:
> I'm not sure if that's true at this point. DialPeer is being scheduled, it may fail. Also, DialPeer may result in dialing not addresses.first(), but some other address.

A `DialPeer` is scheduled for the peer that is potentially replaced if it does not respond, i.e. for `disconnected` (this is a `PeerId`). `peer` is the one who just connected via `address` and is pending insertion (if `disconnected` turns out unresponsive). So I think this is correct. Does that help? If you have suggestions for clarifying the documentation, feel free to open a PR!

Reply:
Thanks for clarifying! I definitely missed that `DialPeer` is issued to `disconnected`, not to freshly connected `peer`. I was wrong, thank you for explanation!